core/src/main/kotlin/com/alecstrong/sql/psi/core/sql.bnf

```diff
@@ -223,13 +224,15 @@ expr ::= ( raise_expr
          | binary_pipe_expr
          | unary_expr
          | literal_expr
-         | other_expr
          | column_expr
          | bind_expr )
 
 other_expr ::= extension_expr
-extension_expr ::= NULL
+extension_expr ::= NULL | multi_column_expr
```

Comment:
in general we want to leave this as NULL to allow for other dialects to override it. Instead you can just inline `multi_column_expr` into `expr` wherever it should take priority

Reply:
Yeah, what do think about moving custom_expr to the top too? This allows you to real overwrite expr, otherwise the grammar is in another sub expr and requires unexpected tokens.

Reply:
yea I think that makes sense? I think either way has drawbacks, when you're highest priority your rule can get successfully parsed into after the first token so it can potentially cause issues downstream, it might be worth testing that the existing dialects dont fail if we move it to the top. Otherwise I agree. It'd be nice if we could just override the `expr` rule though, there might be hacky ways we can get it to work the same way with grammar kit without using the extends stuff

Reply:
@AlecStrong this is new territory for me, when you say inlining `multi_column_expr` into `expr`, what does that mean in this context? is there perhaps some example where inlining is done already that I could take a look at?

Reply:
@sphrak Alec mean this:
```
expr ::= ( other_expr
         | raise_expr
         | case_expr
         | exists_expr
         | paren_expr
         | binary_or_expr
         | binary_and_expr
         | between_expr
         | binary_like_expr
         | is_expr
         | null_expr
         | collate_expr
         | cast_expr
         | function_expr
         | in_expr
         | binary_equality_expr
         | binary_boolean_expr
         | binary_bitwise_expr
         | binary_add_expr
         | binary_mult_expr
         | binary_pipe_expr
         | unary_expr
         | literal_expr
         | column_expr
         | bind_expr
         | multi_column_expr
         )

other_expr ::= extension_expr
extension_expr ::= NULL
multi_column_expr ::= values_expression [ ( LT | LTE | GT | GTE ) values_expression ] {
  pin = 1
}
```
"Just" move the multi_column_expr to expr :)

Reply:
@hfhbd hmm I see thank you, but doing that makes the test break -- unless I have `extension_expr ::= NULL | multi_column_expr` I get:

```sh
java.lang.AssertionError: Test failed because the compile output is missing the errors <[
    Test.s line 17:9 - '(', ')', ',', '.', <binary like operator real>, BETWEEN or IN expected, got ','
]> and unexpectedly has the errors <[
    Test.s line 10:9 - '(', ')', '.', <binary like operator real>, BETWEEN or IN expected, got ','
]>. 
Overall we expected to see the errors <[
    Test.s line 17:9 - '(', ')', ',', '.', <binary like operator real>, BETWEEN or IN expected, got ','
]> but got the errors <[
    Test.s line 10:9 - '(', ')', '.', <binary like operator real>, BETWEEN or IN expected, got ','
]>
```

Reply:
@sphrak The parser is greedy, and `paren_expr` already matches `(id`, but the comma is now unexpected. So you have to move `multi_column_expr` right before `paren_expr`: 
<img width="1920" alt="Bildschirm­foto 2022-07-28 um 23 02 17" src="https://user-images.githubusercontent.com/22521688/181637165-a5be3b7a-5855-452c-8bad-c7b634942f3f.png">

Reply:
@hfhbd oh so the order matters there already, i guess i took u too literally before. let me type it up and push it

Reply:
I guess the next step would be to try and run this with sqldelight and see if everything passes.

Reply:
Yeah, one easy option is to build it locally and copy the artifact into the local repo of sqldelight. 
I don't think, sql-psi has an automatically test during PR using GitHub action to test with latest sqldelight artifacts (or other way around), which would be nice, won't it @AlecStrong?

Reply:
it would be nice! though its okay if sql-psi breaks sqldelight, sometimes thats expected.